controllers/disruption_controller.go

```diff
@@ -673,11 +673,17 @@ func (r *DisruptionReconciler) generatePod(instance *chaosv1beta1.Disruption, ta
 	hostPathDirectory := corev1.HostPathDirectory
 	hostPathFile := corev1.HostPathFile
 
+	// The default TerminationGracePeriodSeconds is 30s. This can be too low for a chaos pod to finish cleaning. After TGPS passes,
+	// the signal sent to a pod becomes SIGKILL, which will interrupt any in-progress cleaning. Setting this to a generous 2 minutes in the pod spec itself,
+	// ensures that whether a chaos pod is deleted directly or by deleting a disruption, it will have time to finish cleaning up after itself.
+	terminationGracePeriod := int64(120)
```

Comment:
As discussed together, can we pass a context to the `Inject` method of injectors so we can cancel the context if we receive the signal during the injection and reduce that termination grace period?

Reply:
After implementation, I decided this was a huge amount of code+complexity for very little perf gain, so I'd like to compromise at just a 1 minute grace period